File: test/Dapr.Actors.AspNetCore.IntegrationTest.App/Startup.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
// ------------------------------------------------------------

using Dapr.Actors.AspNetCore.IntegrationTest.App.ActivationTests;
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;

namespace Dapr.Actors.AspNetCore.IntegrationTest.App
{
    public class Startup
    {
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddScoped<CounterService>();
            services.AddActors(options =>
            {
                options.Actors.RegisterActor<DependencyInjectionActor>();
            });
        }

        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }

            app.UseRouting();

            app.UseEndpoints(endpoints =>
            {
                endpoints.MapActorsHandlers();
            });
        }
    }
}

File: DemoActor.FSharp/Startup.fs
namespace DemoActor.FSharp

open Microsoft.AspNetCore.Builder
open Microsoft.AspNetCore.Hosting
open Microsoft.Extensions.DependencyInjection
open Microsoft.Extensions.Hosting

open DemoActor.FSharp.BankService
open DemoActor.FSharp.BankActor

// Annorances:
//
// 1. Typical F# <-> ASP.NET impedence mismatches
type Startup() =
    member this.ConfigureServices(services: IServiceCollection) =
        services.AddSingleton<BankService>()
                .AddActors(fun options -> options.Actors.RegisterActor<BankActor> |> ignore)

    member this.Configure(app: IApplicationBuilder, env: IWebHostEnvironment) =
        if env.IsDevelopment() then
            app.UseDeveloperExceptionPage() |> ignore

        app.UseRouting()
           .UseEndpoints(fun endpoints ->
                endpoints.MapActorsHandlers() |> ignore
            ) |> ignore
